src/js/viz.js

```diff
@@ -212,16 +207,28 @@ const viz = {
     });
   },
 
-  async drawFavicon(node, x, y) {
-    const offset = this.getSquare().offset,
+  async drawFavicon(node, x, y, radius) {
```

Comment:
@Bilie thanks for this PR. However, I am not convinced about how moving the code to `drawFavicon` function changes anything? Please correct me if I am missing anything obvious.
Moreover, now we have lost the caching functionality and on each call to `drawFavicon`, `imageCanvas` is being set.
Here is the [PR](https://github.com/mozilla/lightbeam-we/pull/201/files) where we implemented the caching functionality. With the moving of the code into the `drawFavicon` function, we now have come back to the non-caching version.

Reply:
From what I can tell this PR caches the network request which is great however In think we can save ourselves from re-rendering the favicon on every load twice.  Currently you are returning the canvas object then scaling it down. I personally think it makes more sense for this to be done inside `convertURIToImageData` as @princiya is mentioning and return the image data of a 64px icon perhaps, then use this to scale down to the right size on the main canvas.

So I think something like the following code can be in  `convertURIToImageData`:
```
    const imageContext = node.imageCanvas.getContext('2d');
    const side = this.getSquare(radius).side * this.scale;

    node.imageCanvas.width = side;
    node.imageCanvas.height = side;
    imageContext.fillStyle = this.canvasColor;
    imageContext.fillRect(0, 0, this.maxSideSize, this.maxSideSize);
    imageContext.drawImage(node.image, 0, 0, this.maxSideSize, this.maxSideSize);

   return imageContext.getImageData(0, 0, this.maxSideSize, this.maxSideSize);
```

Then in `drawIcon` just scale that icon down to the right size.